app/src/main/java/com/github/se/project/ui/components/LessonLocationDisplay.kt

```diff
@@ -23,33 +49,130 @@ fun LessonLocationDisplay(
     lessonTitle: String,
     modifier: Modifier = Modifier
 ) {
-  val location = LatLng(latitude, longitude)
+  var userLocation by remember { mutableStateOf<LatLng?>(null) }
+  var isLocationChecked by remember { mutableStateOf(false) }
+
+  val lessonLocation = LatLng(latitude, longitude)
+
   val cameraPositionState = rememberCameraPositionState {
-    position = CameraPosition.fromLatLngZoom(location, 15f)
+    position = CameraPosition.fromLatLngZoom(lessonLocation, 10f)
   }
 
-  Column(modifier = modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
-    Text(text = "Lesson Location", style = MaterialTheme.typography.titleMedium)
-
-    Card(
-        modifier = Modifier.fillMaxWidth(),
-        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)) {
-          Box(modifier = Modifier.fillMaxWidth().height(200.dp)) {
-            GoogleMap(
-                modifier = Modifier.fillMaxSize(),
-                cameraPositionState = cameraPositionState,
-                properties = MapProperties(isMyLocationEnabled = false),
-                uiSettings =
-                    MapUiSettings(
-                        zoomControlsEnabled = false,
-                        scrollGesturesEnabled = false,
-                        zoomGesturesEnabled = false,
-                        tiltGesturesEnabled = false,
-                        rotationGesturesEnabled = false,
-                    )) {
-                  Marker(state = MarkerState(position = location), title = lessonTitle)
-                }
-          }
+  // Request for location permission launcher
+  LocationPermissionHandler { location ->
+    userLocation = location
+    isLocationChecked = true
+
+    // Update the camera position based on user location availability
+    cameraPositionState.position =
+        if (location != null) {
+          getCameraPositionForBothLocations(
+              userLocation = location, lessonLocation = lessonLocation)
+        } else {
+          CameraPosition.fromLatLngZoom(lessonLocation, 10f)
```

Comment:
Great idea to offer a default. Not all users will be comfortable giving their location, so not making it mandatory removes a possible barrier to some!